scheduling/taskmetadata.go

```diff
@@ -186,6 +194,78 @@ func (m *TaskMetadataManager) CreateTaskRun(name string, taskID TaskID, trigger
 	return metadata, nil
 }
 
+func (m *TaskMetadataManager) latestTaskRun(runs TaskRuns) (TaskRunID, error) {
+
+	var latestTime time.Time
+	var latestRunIdx int
+	for i, run := range runs.Runs {
+		if i == 0 {
+			latestTime = run.DateCreated
+			latestRunIdx = i
+		} else if run.DateCreated.After(latestTime) {
+			latestTime = run.DateCreated
+			latestRunIdx = i
+		}
+	}
+	return runs.Runs[latestRunIdx].RunID, nil
+}
+
+// GetLatestRun is not guaranteed to be completely accurate. This function should only
+// be used for visual purposes on the Dashboard and CLI rather than internal business logic
```

Comment:
how do we guarantee a dev won't reach to use this. maybe rename like `GetLatestRunDeprecated(...)`?

Reply:
So its not deprecated, but its just not safe to use for things other than user facing applications where calling it a few ms before or after makes a difference